### PR TITLE
Small optic improvements to the module chooser

### PR DESCRIPTION
**Pull request by MrEbbinghaus**
FontAwesome can now be used as module-icons. (e.g. config.json entry: "icon_fa" : "fa-twitter")
You can find the FontAwesome Icons [here](http://fortawesome.github.io/Font-Awesome/icons/).
Addition: changed the config.json of "twitter"

Transition effects on the module links (ease-out 300ms) 
-> defined in "boxes.css.opacity-trans"

fixed Issue #27 by removing the failed icon with an inline JavaScript code

minor changes:
added breaks to "css/font-awesome.min.css" for more readability 
changed the space between head an list of the module-chooser from "padding" to "margin"
set z-index from inner-site-wrapper to "-1"

### CODE REVIEW

File: js/core.coffee [hunk 1]
```diff
@@ -135,29 +135,34 @@ toggle_highlighted_boxes = (thisBox) ->
...
 
     content += "<ul>"
     for module in modules
         if (module.charAt 0) != '.'
 
-            # Assign values from the correlated config.json
             config = load_conf path.join(modpath, module)
 
-            content += "<li class='module-entry'><a class='module-single' href='#' name='#{module}'"
+            content += "<li class='module-entry'><a class='module-single opacity-trans' href='#' id='#{module}' name='#{module}'"
```
Comment (MrEbbinghaus) [inline js/core.coffee:146]: `id='#{module}'` not necassary

File: modules/twitter/config.json [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
-    "icon" :"/img/icon.png"
-
+    "name" : "Twitter",
+    "description" : "Display your Twitter Stream inside trickle. Needs you to authorize the App to Twitter",
+    "type" : "stream",
+    "options" : ["Twitter-Name"],
+    "hook" : "twitter.js",
+    "version" : 0.1,
+    "color": "#55ACEE",
+    "icon" : "/img/icon.pn",
```
Comment (MrEbbinghaus) [inline modules/twitter/config.json:9]: delete `"icon_fa" : "fa-twitter"` to test that core.js can handle an invalid path

**Comment by n2o** [comment]
Some things:
- I chose the minified version of fontawesome to update it easier. Changes can bei made in the other css files
- there's inline CSS. Avoid it where it is possible
- Write JSON this way:

``` json
{
    "keyword": "value"
}
```

and not 

``` json
{
    "keyword" : "value"
}
```

File: css/bootstrap.cover.css [hunk 3]
```diff
@@ -45,8 +45,10 @@ body {
           box-shadow: inset 0 0 100px rgba(0,0,0,.5);
 }
 .site-wrapper-inner {
+    z-index: -1;
```
Comment (n2o) [inline css/bootstrap.cover.css:48]: Wrong indention

File: css/boxes.css [hunk 4]
```diff
@@ -9,13 +9,30 @@ div#boxes div.box-content {
...
+.module-single{
     padding: 0.5em;
-    min-width: 200px;
+    max-width: 100%;
+    opacity: 1;
+}
+.module-single:hover{
+    text-decoration: none;
+
+    /*not tested with light backgrounds!*/
```
Comment (n2o) [inline css/boxes.css:21]: useless comment. there are no light backgrounds

File: js/core.coffee [hunk 5]
```diff
@@ -135,29 +135,34 @@ toggle_highlighted_boxes = (thisBox) ->
...
 
+            # assign the color to the background
             if bcolor != "" && bcolor
                 content += " style='background-color: #{bcolor};'"
 
             content += ">"
 
-            if icon then content += "<img class='icon' src='#{icon}' alt='#{module}'> "
+            # decide which icon has to be showed
+            if icon_fa then content += "<span class='fa #{icon_fa}'></span>&nbsp;"
```
Comment (n2o) [inline js/core.coffee:162]: Inline IFs do not match our style

File: css/boxes.css [hunk 6]
```diff
@@ -9,13 +9,30 @@ div#boxes div.box-content {
...
+    opacity: 0.7;
+}
+
+/*opacity transition*/
+.opacity-trans{
+    -webkit-transition:opacity 300ms ease-out;
```
Comment (n2o) [inline css/boxes.css:27]: We already have transitions. Use the same time as we do (currently `.25s`)

File: css/boxes.css [hunk 7]
```diff
@@ -9,13 +9,30 @@ div#boxes div.box-content {
...
+    -moz-transition:opacity 300ms ease-out;
+    -o-transition:opacity 300ms ease-out;
+    transition:opacity 300 ease-out;
```
Comment (n2o) [inline css/boxes.css:30]: You forgot the Unit 300 **ms**

File: css/boxes.css [hunk 8]
```diff
@@ -9,13 +9,30 @@ div#boxes div.box-content {
 .module-entry {
     margin: 1em;
 }
-.module-single {
+
+.module-single{
```
Comment (n2o) [inline css/boxes.css:13]: Use a space before the brackets

File: css/boxes.css [hunk 9]
```diff
@@ -9,13 +9,30 @@ div#boxes div.box-content {
...
     padding: 0.5em;
-    min-width: 200px;
+    max-width: 100%;
+    opacity: 1;
+}
+.module-single:hover{
```
Comment (n2o) [inline css/boxes.css:18]: Space before brackets...

File: css/boxes.css [hunk 10]
```diff
@@ -9,13 +9,30 @@ div#boxes div.box-content {
...
+    text-decoration: none;
+
+    /*not tested with light backgrounds!*/
+    opacity: 0.7;
+}
+
+/*opacity transition*/
```
Comment (n2o) [inline css/boxes.css:25]: Wrap comments in one space:
`/* Insane comment */`

File: css/boxes.css [hunk 11]
```diff
@@ -9,13 +9,30 @@ div#boxes div.box-content {
...
+.opacity-trans{
+    -webkit-transition:opacity 300ms ease-out;
+    -moz-transition:opacity 300ms ease-out;
+    -o-transition:opacity 300ms ease-out;
+    transition:opacity 300 ease-out;
 }
 
 .icon{
```
Comment (n2o) [inline css/boxes.css:33]: Space before bracket

File: modules/twitter/config.json (hunk 2 above)
Comment (n2o) [inline modules/twitter/config.json:9]: Typo. Must be `icon.png`

**Pull request merged**